src/reporting/Market.sol

```diff
@@ -0,0 +1,604 @@
+    }
+
+    function getFinalPayoutDistributionHash() public constant returns (bytes32) {
+        return finalPayoutDistributionHash;
+    }
+
+    function getPayoutDenominator() public constant returns (uint256) {
+        return payoutDenominator;
+    }
+
+    function getDenominationToken() public constant returns (Cash) {
+        return cash;
+    }
+
+    function getMarketCreatorSettlementFeeInAttoethPerEth() public constant returns (uint256) {
+        return feePerEthInAttoeth;
+    }
+
+    function getMaxDisplayPrice() public constant returns (int256) {
+        return maxDisplayPrice;
+    }
+
+    function getMinDisplayPrice() public constant returns (int256) {
+        return minDisplayPrice;
+    }
+
+    function getCompleteSetCostInAttotokens() public constant returns (uint256) {
+        return uint256(maxDisplayPrice.sub(minDisplayPrice));
+    }
+
+    function getTopic() public constant returns (bytes32) {
+        return topic;
+    }
+
+    function shouldCollectReportingFees() public constant returns (bool) {
+        return !getBranch().isContainerForShareToken(cash);
+    }
+
+    function isDoneWithAutomatedReporters() public constant returns (bool) {
+        return automatedReportReceived || block.timestamp > getAutomatedReportDueTimestamp();
+    }
+
+    function isDoneWithLimitedReporters() public constant returns (bool) {
+        return getReportingState() > ReportingState.LIMITED_REPORTING;
+    }
+
+    function isDoneWithAllReporters() public constant returns (bool) {
+        return getReportingState() > ReportingState.ALL_REPORTING;
+    }
+
+    function isFinalized() public constant returns (bool) {
+        return finalPayoutDistributionHash != bytes32(0);
+    }
+
+    function getFinalizationTime() public constant returns (uint256) {
+        return finalizationTime;
+    }
+
+    function isInAutomatedReportingPhase() public constant returns (bool) {
+        return getReportingState() == ReportingState.AUTOMATED_REPORTING;
+    }
+
+    function isInAutomatedDisputePhase() public constant returns (bool) {
+        return getReportingState() == ReportingState.AUTOMATED_DISPUTE;
+    }
+
+    function isInLimitedReportingPhase() public constant returns (bool) {
+        return getReportingState() == ReportingState.LIMITED_REPORTING;
+    }
+
+    function isInLimitedDisputePhase() public constant returns (bool) {
+        return getReportingState() == ReportingState.LIMITED_DISPUTE;
+    }
+
+    function isInAllReportingPhase() public constant returns (bool) {
+        return getReportingState() == ReportingState.ALL_REPORTING;
+    }
+
+    function isInAllDisputePhase() public constant returns (bool) {
+        return getReportingState() == ReportingState.ALL_DISPUTE;
+    }
+
+    function isContainerForReportingToken(ReportingToken _shadyToken) public constant returns (bool) {
+        if (address(_shadyToken) == NULL_ADDRESS) {
+            return false;
+        }
```

Comment:
We can remove this check, since the next line will throw in Solidity anyway (test to validate, but I believe calling a method on address 0 fails in Solidity).